### PR TITLE
Python edits to Command documentation

### PR DESCRIPTION
**Pull request by mbardoe**
In response to requests for Python additions to the Command Based robot documentation I have tried to incorporate the appropriate Python code and links for some of the files. You can see this CD post [https://www.chiefdelphi.com/t/robotpy-2024-beta/443117] (https://www.chiefdelphi.com/t/robotpy-2024-beta/443117) #2645. This is my first time with frc-docs contributions so let me know the ways that this can be improved.

### CODE REVIEW

**Comment by sciencewhiz** [comment]
The links to python API docs can be simplified to the following (for the Driverstation as example)

```
:py:class:`Python <robotpy:wpilib.DriverStation>`
```

**Comment by mbardoe** [comment]
> The links to python API docs can be simplified to the following (for the Driverstation as example)
> 
> ```
> :external:py:class:`Python <robotpy:wpilib.DriverStation>`
> ```

I am struggling to make these work. Most of my links are to the `commands2` library rather than straight wpilib. Also, many of the links go to specific methods of the classes, and I am not sure how to link to those.

**Comment by virtuald** [comment]
Because it's new, nobody added it to intersphinx. You can link to them using commands2 instead of robotpy if you use the commit in https://github.com/wpilibsuite/frc-docs/pull/2657

**Comment by sciencewhiz** [comment]
#2657 has been merged, so you can merge main and use it now.

File: source/docs/software/commandbased/command-compositions.rst [hunk 1]
```diff
@@ -19,6 +19,11 @@ Most importantly, however, command compositions are themselves commands - they e
       // Will run fooCommand, and then a race between barCommand and bazCommand
       button.OnTrue(std::move(fooCommand).AndThen(std::move(barCommand).RaceWith(std::move(bazCommand))));
 
+   .. code-block:: python
+
+      # Will run fooCommand, and then a race between barCommand and bazCommand
+      button.OnTrue(fooCommand.andThen(barCommand.raceWith(bazCommand)))
```
Comment (Starlight220) [inline source/docs/software/commandbased/command-compositions.rst:25]: Casing is inconsistent here -- what do the actual functions use?

**Review by sciencewhiz — APPROVED** [review]

**Pull request merged**